src/twisted/logger/test/test_global.py

```diff
@@ -356,3 +357,16 @@ def test_warningsModule(self):
                 filename=__file__, lineno=2,
             )]
         )
+
+
+    def test_failuresAppendTracebacks(self):
+        """
+        The string resulting from a logged failure contains a traceback.
+        """
+        f = Failure(Exception("this is not the behavior you are looking for"))
+        log = Logger(observer=self.publisher)
+        log.failure('a failure', failure=f)
+        msg = self.errorStream.getvalue()
+        self.assertIn('a failure', msg)
```

Comment:
This (and what follows) should _not_ be a unicode literal :)

That's because [FileLogObserver encodes unicode messages to UTF-8 when it's given a binary file](https://github.com/twisted/twisted/blob/104394ddb76775c57c18897160a5ad055617b51a/src/twisted/logger/_file.py#L33-L37), and [it is on Python 2](https://github.com/jlitzingerdev/twisted/blob/a0a1c7c2bd7b7a2f396ea6ddb0453c131ccec4fc/src/twisted/logger/test/test_global.py#L65).

Reply:
@markrwilliams Am I correct in interpreting that this code should remain as-is, to exercise the different paths for a native string on Python2/Python3?  Or, did you want me to ensure the encoding was forced on both?

Reply:
Sorry for being unclear!  This was more of a note to myself confirming that this code is correct to use a native string.